### PR TITLE
[10647] Provide an alternative path for robustlock files under QNX <master>

### PR DESCRIPTION
**Pull request by IkerLuengo**
This is a port of #1766 to master

### CODE REVIEW

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
Change [this line](https://github.com/eProsima/Fast-DDS/blob/0195e51ab7ef9fe2d7039acbcf66dd1c03e3fc9d/src/cpp/rtps/DataSharing/DataSharingPayloadPool.hpp#L26) to include `SharedDir.hpp` instead of `RobustExclusiveLock.hpp`

**Review by MiguelCompany — DISMISSED** [review]

**Comment by MiguelCompany** [comment]
@asobhy-qnx Would you be so kind to check if this works on QNX?

**Comment by asobhy-qnx** [comment]
Sure

**Comment by asobhy-qnx** [comment]
got this error:
[RTPS_MSG_OUT Error] error: couldn't generate segment_name -> Function init
 I have not looked at the source yet will let you know when I figure out what is wrong

File: src/cpp/utils/shared_memory/SharedDir.hpp [hunk 1]
```diff
@@ -67,6 +73,37 @@ class SharedDir
...
+            {
+                std::string err("get_file_path() ");
+                err = err + strerror(errno);
+                throw std::runtime_error(err);
+            }
+        }
+        else
+        {
+            // directory exists do nothing
+        }
```
Comment (asobhy-qnx) [inline src/cpp/utils/shared_memory/SharedDir.hpp:101]: this function does not return anything for when ```__QNXNTO__``` is defined.
add the following line:

```
return (std::string(defaultdir) + "/" + filename) ;
```

**Review by asobhy-qnx — CHANGES_REQUESTED** [review]

File: src/cpp/utils/shared_memory/SharedDir.hpp (hunk 1 above)
Comment (MiguelCompany) [reply src/cpp/utils/shared_memory/SharedDir.hpp:101]: @asobhy-qnx Thanks for the suggestion! I missed that on my review 🤦 

I have added the necessary change on 061a563. Please check that it works.

**Comment by MiguelCompany** [comment]
@richiprosima Please check style

File: src/cpp/utils/shared_memory/SharedDir.hpp (hunk 1 above)
Comment (asobhy-qnx) [reply src/cpp/utils/shared_memory/SharedDir.hpp:101]: Works, thank you.

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**